### ISSUE TITLE
barcode scanner

### ISSUE THREAD

**Issue by LiorBenAmi** [TechnionYP5777/SmartCity-Market#43]
* Investigate and learn the issue of  barcode scanner technologies
* Think of solution that can be placed on the smart cart.
Restrictions:
1. linux OS
2. java directories

next - implement the scanner

**Comment by LiorBenAmi** [comment]
#39

**Comment by yossigil** [comment]
Redefine the isssue so that it is clear what is expected, and how we will know it is done.

**Comment by shimonAzulay** [comment]
@LiorBenAmi there is any progress with this issue?

**Comment by LiorBenAmi** [comment]
@shimonAzulay  In progress

**Comment by LiorBenAmi** [comment]
Search results:
* Allot of products to: scan/print and decode/encode GS1-128 bar-codes.
* but, most of them with payment
* free/open source:
** [zxing](https://github.com/zxing/zxing)
**[zbar](http://zbar.sourceforge.net/index.html) - scanner only
**[barcode4j](http://barcode4j.sourceforge.net/) - generator only
**[zint](https://sourceforge.net/projects/zint/)
**[libdmtx](https://sourceforge.net/projects/libdmtx/)

next: choose and implement

**Comment by LiorBenAmi** [comment]
- [zbar](http://zbar.sourceforge.net/index.html) - app - the code in C
- [zint](https://sourceforge.net/projects/zint/) - in python/pearl
- [libdmtx](https://sourceforge.net/projects/libdmtx/) - the code is in C

remain only with [zxing](https://github.com/zxing/zxing) - has maven configuration. need to learn how to use (find good tuturials)

**Comment by IdanAtias** [comment]
@yossigil @AviadCo 
Do u happen to know if the lab holds a scanner (like the one in the library for scanning the barcodes on the books before they are loaned to students) that we can borrow?

If yes, do you think it has some kind of API that can use us for scanning the barcodes from within our java code?

I'm asking you because till now, all we found is ANDROID-BASED features to scan barcode with smart-phones, and we build a desktop app...

Any advice could help here...
Thanks!

**Comment by AviadCo** [comment]
Hi  @IdanAtias 

I've just read about barcode scanner for an hour..

As far as i found, there are many barcode interpreters from image files but none from camera.
i think we should try to choose one interpreter from image file to barcode (there are many java code projects for those this, and i remember that @LiorBenAmi said he succeeded doing so) - when this works, we can use the PC Web camera to take a picture and convert this to am image file. The only thing left is to send the picture to the interpreter. it's a long shot, and i am not pretty sure it will work (the pc camera doesn't have the quality of phone one, but it's the best we can do).

For now i don't have better idea (other than to buy real scanner)...

**Comment by yossigil** [comment]
sorry, cant help here. Maybe TOM at SSDL?

**Comment by shimonAzulay** [comment]
Hi guys! 
So for now as i understand from you, we couldn't find any end to end camera to bar code integer number parser. So we could build it by ourselves  (Like we love to do).
As @AviadCo said we can use service for parsing image to int number. i found this http://webcam-capture.sarxos.pl/ in a quick search in the internet. It can gives us image parsing. So maybe we can combine the two services until we find other solution. What do you think ?

**Comment by shimonAzulay** [comment]
@yossigil @zvili @idabran 
Hi ! 
So @AviadCo suggested another solution to our (life) problem. If there isn't Bar code scanner with an API that we can borrow from the Technion, We can (and again as we love to do) create it by ourselves. We will use our Android  smart mobile phone as a bar code scanner. There is an implementation that we are familiar with (Zxing lib) that parse bar code from Camera to java code int number. We can create simple Android application that will run this app, When a product's bar code is scanned, the application will trigger our (Main project -Smart Market) app. As for sending data from the smart phone to our Win app, i am sure we could find simple solution. 

Bottom line: 
We have 3 solutions:
1. Borrow bar code scanner with API that we can use.
2. Implement camera -> image -> int scanner (with use of 3rd party libs for camera -> image parsing and image -> int parsing) as i described in my previous comment.
3. Use our Android smart phone to parse the bar code and send it to our Win Smart Market app.

**Comment by yossigil** [comment]
@idabran please help.

Zvi is OOO and so am I. 
@IdanAtias please call @idabran in person if he doesn't respond.

**Comment by IdanAtias** [comment]
Thanks.
@AviadCo talked with him.
Aviad, maybe you can write here the suggested solution...

**Comment by AviadCo** [comment]
I spoke with @idabran .

I updated him regarding the issue of the barcode scanner. (he also thought it would be harder to picture the barcode which will be clear enough to process)
Itai allowed us that to use phone to scan the barcode and send it to the computer to complete the client flow. 

@shimonAzulay  will implement the barcode scanner (Please open issue if none opened before regarding barcode scanner)

I'm closing this issue.
